### ISSUE TITLE
Throw an error when using @Property with an @Entity-decorated type

### ISSUE THREAD

**Issue by rcfox** [mikro-orm/mikro-orm#3807]
**Is your feature request related to a problem? Please describe.**

If you accidentally use `@Property` instead of one of the relationship decorators (`@ManyToOne`, etc.), no errors are thrown until you actually attempt to use the field in a query. The error you get there is confusing: I got "Invalid query condition" in one situation and in another, where there was a nested query, it attempted to find the field on the parent entity.

**Describe the solution you'd like**
Ideally, this issue would be detected at compile-time. I'm not sure if that is feasible though.

It should at least trigger an error when using one of the schema/migration CLI tools. Preferably with some guidance on how to fix the issue.

**Describe alternatives you've considered**
If this is difficult to implement within the library, perhaps it would be possible as an eslint rule?

**Additional context**
N/A

**Comment by B4nan** [comment]
Can you provide exact entity definition that should throw?

**Comment by rcfox** [comment]
Entities:
```
@Entity()
export class Foo {
  @PrimaryKey()
  id!: number;

  @Property()
  name!: string;
}

@Entity()
export class Bar {
  @PrimaryKey()
  id!: number;

  @Property() // Should be @ManyToOne()
  test!: Foo;
}
```

Use:
```
const foo = em.create(Foo, { name: 'Foo' });
em.create(Bar, { test: foo });
await em.flush();

await em.find(Bar, { test: { name: 'Foo' } }); // DriverException: Invalid query condition: { test: { name: 'Foo' } }
```

**Comment by B4nan** [comment]
There is no way to test this via TS, but sure we can have a runtime validation after the init phase.

**Comment by liang799** [comment]
Hi, may I know if there anyone is currently working on resolving this issue?

**Comment by B4nan** [comment]
This is actually quite complicated, I gave it a try two times already :] I started making some adjustments to v6 so this can be done more easily. But maybe you will find some easy way I totally missed.

If you are looking for some low-hanging fruit, the other "good first issue" (#3855) is much simpler to solve.

**Comment by liang799** [comment]
Noted with thanks

**Comment by B4nan** [comment]
Closing as implemented in v6 (8d5bd84176e5799a700fe3c1bfbcbc169b79d083)
